clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntityFactory.java

```diff
@@ -258,11 +297,72 @@ private VoltronReadWriteLock createAccessLockFor(String entityIdentifier) {
 
   private EntityRef<EhcacheClientEntity, ClusteredTierManagerConfiguration> getEntityRef(String identifier) {
     try {
-      return connection.getEntityRef(EhcacheClientEntity.class, EhcacheEntityVersion.ENTITY_VERSION, identifier);
+      return connection.getEntityRef(EhcacheClientEntity.class, ENTITY_VERSION, identifier);
     } catch (EntityNotProvidedException e) {
       LOGGER.error("Unable to get clustered tier manager for id {}", identifier, e);
       throw new AssertionError(e);
     }
   }
 
+  public ClusteredTierClientEntity fetchOrCreateClusteredStoreEntity(UUID clientId, String clusterTierManagerIdentifier,
+                                                                     String storeIdentifier, ServerStoreConfiguration clientStoreConfiguration,
+                                                                     boolean autoCreate) throws EntityNotFoundException, CachePersistenceException {
+    EntityRef<ClusteredTierClientEntity, ClusteredTierEntityConfiguration> entityRef;
+    try {
+      entityRef = connection.getEntityRef(ClusteredTierClientEntity.class, ENTITY_VERSION, clusterTierManagerIdentifier + storeIdentifier);
```

Comment:
The entity API provides the guarantee that only one caller will succeed. We do not need anything more here. The locking at the `CacheManager` level is required for maintenance but that is not true for `Cache` level lifecycle.